robottelo/commands/ui.py

```diff
@@ -0,0 +1,127 @@
+# coding: utf-8
+
+import click
+import import_string
+
+from functools import partial
+from inspect import getmembers
+from manage.cli import create_shell
+from nailgun import entities
+from robottelo.cli import factory as cli_factory
+from robottelo.config import settings
+from robottelo.constants import UI_CRUD
+from robottelo.ui.browser import browser as selenium_browser
+from robottelo.ui.browser import DockerBrowser
+from robottelo.ui import factory as ui_factory
+from robottelo.ui.session import Session
+
+
+settings.configure()
+
+
+def _get_browser(browser=None):
+    """Gets a new instance of a browser to interact"""
+
+    browser = browser or settings.browser
+    if browser == 'docker':
+        _docker_browser = DockerBrowser()
+        _docker_browser.start()
+        _browser = _docker_browser.webdriver
+    elif browser == "selenium":
+        _browser = selenium_browser()
+    else:
+        raise NotImplementedError(
+            "this shell only supports docker and selenium")
+
+    _browser.maximize_window()
+    _browser.get(settings.server.get_url())
+    return _browser
+
+
+class Storage(object):
+    """Turns a dict in to attribute based object"""
+    def __init__(self, *args, **kwargs):
+        """takes a dict or attrs and sets as attrs"""
+        super(Storage, self).__init__()
+        for item in args:
+            kwargs.update(item)
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+
+def _import_ui_crud():
+    """Imports all UI crud related classes to shell scope"""
+    base_module = "robottelo.ui.{0}"
+    return {
+        name.split('.')[-1]: import_string(base_module.format(name))
+        for name in UI_CRUD
+    }
+
+
+@click.command()
+@click.option('--host', required=False, default=None,
+              help="satellite host name e.g:'foo.bar.com'")
+@click.option('--browser', required=False, default=None,
+              help='selenium or docker (defaults to properties file)')
+@click.argument('entity', required=False, default=None)
+def browse(entity, browser, host):
+    """Opens a page in defined browser for interaction:\n
+        example: $ manage ui browse activationkey\n
+                 Opens a browser in ActivationKey scope and gives you
+                 interactive shell to play with the page\n
+
+    """
+    if host:
+        settings.server.hostname = host
+    current_browser = _get_browser(browser)
+    with Session(current_browser) as session:  # noqa
+
+        ui_crud = _import_ui_crud()
+        ui_entities = {
```

Comment:
Best document this... just to make sure anyone can maintain the code later on :)